test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/ObjectsTest.scala

```diff
@@ -94,6 +94,30 @@ class ObjectsTest {
     assertEquals("abc", ju.Objects.requireNonNull("abc", ""))
   }
 
+  @Test def requireNonNullWithMsgSupplier(): Unit = {
+    val message =  "All is well!"
+
+    val successSupplier = new ju.function.Supplier[String] {
+          def get(): String = message
+      }
+
+    val failureSupplier = new ju.function.Supplier[String] {
+        def get(): String = throw new AssertionError(
+            "Objects.requireNonNull() should not have called Supplier")
+      }
+
+    try {
+      ju.Objects.requireNonNull(null, successSupplier)
+      fail("Expected NullPointerException, got success")
+    } catch {
+      case e: NullPointerException => assertEquals(message, e.getMessage())
```

Comment:
I believe you were looking for `AssertThrows.expectThrows`, to assert that an exception is thrown, and additionally get the instance to check its message in a subsequent `assertEquals`.